src/main/java/mate/academy/spring/security/CustomUserDetailsService.java

```diff
@@ -0,0 +1,33 @@
+package mate.academy.spring.security;
+
+import java.util.Optional;
+import lombok.AllArgsConstructor;
+import mate.academy.spring.model.User;
+import mate.academy.spring.service.UserService;
+import org.springframework.security.core.userdetails.User.UserBuilder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+@AllArgsConstructor
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserService userService;
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+        Optional<User> optionalUser = userService.findByEmail(email);
+        UserBuilder builder;
+        if (optionalUser.isPresent()) {
+            builder = org.springframework.security.core.userdetails.User.withUsername(email);
+            builder.password(optionalUser.get().getPassword());
+            builder.roles(optionalUser.get().getRoles()
+                    .stream()
+                    .map(role -> role.getName().toString())
```

Comment:
I'm not sure that you tested it
```suggestion
                    .map(role -> "ROLE_" + role.getName().name())
```